### PR TITLE
feat: execute template methods directly

### PR DESCRIPTION
**Pull request by ryan-timothy-albert**
Allow template methods to be executed directly from easy template

### CODE REVIEW

File: engine.go [hunk 1]
```diff
@@ -160,6 +165,61 @@ func (e *Engine) RunScript(scriptFile string, data any) error {
 	return nil
 }
 
+// MethodExecutor returns an execution function that enables calls to global template methods from easytemplate.
+func (e *Engine) MethodExecutor(scriptFile string, data any) (func(fnName string, args ...interface{}) (interface{}, error), error) {
```
Comment (TristanSpeakEasy) [inline engine.go:169]: I was actually just thinking of having a `Run` that takes a method name and then someone would call `RunScript` first followed by `Run` to run additional methods in the environment setup by `RunScript`.

That would maybe simplify the implementation here

Comment (ryan-timothy-albert) [reply engine.go:169]: We could do that but don't we need to do this in RunScript which we don't right now

result := esbuild.Transform(string(script), esbuild.TransformOptions{
Target:    esbuild.ES2015,
Loader:    esbuild.LoaderTS,
Sourcemap: esbuild.SourceMapExternal,
})

Comment (ryan-timothy-albert) [reply engine.go:169]: https://github.com/speakeasy-api/easytemplate/blob/main/engine.go#L156

I'm not sure what the implications of adding that for every RunScript would be.

**Review by TristanSpeakEasy — APPROVED** [review]
preapproving based on the changes we discussed in a call happening

**Pull request merged**